pkg/common/upgrade/managed.go

```diff
@@ -75,12 +82,25 @@ func TriggerManagedUpgrade(h *helper.H) (*configv1.ClusterVersion, error) {
 
 	// Create Pod Disruption Budget test workloads if desired
 	if viper.GetBool(config.Upgrade.ManagedUpgradeTestPodDisruptionBudgets) {
-		err = createPodDisruptionBudgetWorkloads(h)
+		err = createManagedUpgradeWorkload(pdbWorkloadName, pdbWorkloadDir, h)
 		if err != nil {
 			return cVersion, fmt.Errorf("unable to setup PDB workload for upgrade: %v", err)
 		}
 	}
 
+	// Create Node Drain test workloads if desired
+	if viper.GetBool(config.Upgrade.ManagedUpgradeTestNodeDrain) {
+		err = createManagedUpgradeWorkload(drainWorkloadName, drainWorkloadDir, h)
+		if err != nil {
+			return cVersion, fmt.Errorf("unable to setup node drain test workload for upgrade: %v", err)
+		}
+
+		// Delete node drain workload so that removal of pod stuck in terminating state will be tested by managed-upgrade-operator
+		log.Printf("Deleting workload %s to test node drain by upgrade", drainWorkloadName)
+		h.Kube().AppsV1().Deployments(h.CurrentProject()).Delete(context.TODO(), drainWorkloadName, metav1.DeleteOptions{})
```

Comment:
IMO we don't need this to occur. This will and should happen naturally as a consequence of the cluster upgrade, rather than be initiated by e2e. (much as it would for normal workloads)

Reply:
Thanks Matt !! Agreed.. Have removed the delete workload stuff to let MUO handle it.. Have rebased and add your change..